docs/utilities/batch.md

```diff
@@ -450,6 +450,18 @@ We can automatically inject the [Lambda context](https://docs.aws.amazon.com/lam
 --8<-- "examples/snippets/batch/accessLambdaContext.ts"
 ```
 
+### Working with full batch failures
+
+By default, the `BatchProcessor` will throw a `FullBatchFailureError` if all records in the batch fail to process, we do this to reflect the failure in your operational metrics.
+
+In some cases, for example such as when working with small batches or when using errors as flow control mechanism, this behavior might not be desired and end up negatively impacting the concurrency of your function.
```

Comment:
It took me some time to understand that "small batches" refers to the size of the batch and not the amount of messages I'm sending to SQS and processing. I think it makes sense to add "e.g 10 messages at a time".

**My idea (need to improve the english):**
In some cases, for example when working with small batches (e.g 10 message batch size) or when using errors as a flow control mechanism, if your function encounters exceptions (code failure), Lambda will scale down the concurrency, potentially causing performance problems. See the [official documentation ](https://docs.aws.amazon.com/lambda/latest/dg/services-sqs-errorhandling.html#services-sqs-backoff-strategy)for more details.

And what about adding this blog post? https://aws.amazon.com/pt/blogs/compute/introducing-faster-polling-scale-up-for-aws-lambda-functions-configured-with-amazon-sqs/

Reply:
OK, I have rephrased the section to make it more clear (I think) and in line with your suggestion.

I included the link to the documentation (thank you!) but opted to not include the one to the blog post, because the details about how errors impact scaling are just in a section that sets the context of the launch - which is itself the focus of the post.

Finally, I used the term `error` instead of `exception` because it's the most idiomatic for Node.js, and also I didn't include the code failure sentence because it kind of narrows down the meaning too much imo.

In my mental model, a code failure is a runtime unhandled exception/error, aka one that comes from a fault or bug in the code. For the type of cases I'm describing there, errors might be thrown intentionally.

Reply:
This is so much better than before!

> In my mental model, a code failure is a runtime unhandled exception/error, aka one that comes from a fault or bug in the code. For the type of cases I'm describing there, errors might be thrown intentionally.

I completely agree with that!